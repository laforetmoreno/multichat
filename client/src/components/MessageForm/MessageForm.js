import React from 'react';
import { func } from 'prop-types';

import Input from '../Input';
import Button from '../Button';
import Form from '../Form';

import styles from './MessageForm.module.scss';

const RoomForm = ({ onSubmit, onLogout, ...props }) => (
  <Form onSubmit={onSubmit} className={styles.wrapper}>
    <Input name="message" className={styles.input} placeholder="Digite uma mensagem" {...props} />
    <div className={styles.btnsContainer}>
      <Button onClick={onSubmit} className={styles.btn}>
        Enviar
      </Button>
      <Button onClick={onLogout} color="warning" className={styles.btn}>
        Deslogar
      </Button>
    </div>
  </Form>
);

RoomForm.propTypes = {
  onSubmit: func,
  onLogout: func,
};

export default RoomForm;
