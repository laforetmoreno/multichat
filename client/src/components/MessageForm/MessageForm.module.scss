.wrapper {
  width: 100%;
  display: flex;

  @media (max-width: 767px) {
    flex-direction: column;
  }

  .input {
    margin-right: 10px;
  }

  .btnsContainer {
    width: 100%;
    display: flex;
    justify-content: space-between;

    @media (max-width: 767px) {
      margin-top: 10px;
    }

    .btn {
      width: 48%;
    }
  }
}
