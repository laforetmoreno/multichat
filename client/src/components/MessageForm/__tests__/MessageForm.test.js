import React from 'react';
import MessageForm from '../MessageForm';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('<MessageForm />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MessageForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('testing events', async () => {
    const { getByTestId } = render(<MessageForm />)
      const fieldInput = await waitForElement(() => getByTestId('input'))
      const value = 'test value'
      fireEvent.change(fieldInput, { target: { value } })

      expect(fieldInput.value).toEqual(value)
  });
});