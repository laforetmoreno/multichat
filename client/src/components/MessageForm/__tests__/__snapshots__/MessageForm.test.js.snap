// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageForm /> renders correctly 1`] = `
<form
  className="wrapper"
>
  <input
    className="wrapper input"
    data-testid="input"
    name="message"
    placeholder="Digite uma mensagem"
  />
  <div
    className="btnsContainer"
  >
    <button
      className="btn primary"
      data-testid="btn"
    >
      Enviar
    </button>
    <button
      className="btn warning"
      data-testid="btn"
    >
      Deslogar
    </button>
  </div>
</form>
`;
